app/validators/routing/new_when_rule_validator.py

```diff
@@ -68,23 +72,14 @@ def validate(self):
 
     def validate_rule(self, rule):
         operator_name = next(iter(rule))
-        argument_types = []
-        for argument in rule[operator_name]:
-            if isinstance(argument, dict) and any(
-                operator in argument for operator in ALL_OPERATORS
-            ):
-                argument_type = self.validate_rule(argument)
-            elif isinstance(argument, dict) and "source" in argument:
-                argument_type = resolve_value_source_json_type(
-                    argument, self.questionnaire_schema.answers_with_context
-                )
-            else:
-                argument_type = python_type_to_json_type(type(argument).__name__)
-            argument_types.append(argument_type)
+        argument_types = self._get_argument_types_for_operator(rule, operator_name)
```

Comment:
Not a blocker, this could pass in the actual operand as `rule[operator_name]` since those are not needed indivudally.